README.md

```diff
@@ -1,2 +1,166 @@
+[codecov-url]:https://codecov.io/gh/moxystudio/next-common-files
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/next-common-files/master.svg
+
 Next.js plugins that insert loaders for common files.
+
+Projects developed with Next.js have to manually insert rules in the configuration file for handling file types that are potentially very common across multiple projects.
+
+These plugins quicken the initial setup of a project by removing that effort from the process, demanding less time from the developer.
+
+
+## Installation
+
+```sh
+$ npm install --save-dev @moxy/next-common-files
+```
+
+
+## Usage
+
+For single usage:
+
+```js
+// next.config.js
+const { withRasterImages } = require('@moxy/next-common-plugins');
+
+module.exports = withRasterImages({
+	/* options */
+});
+```
+
+For using multiple plugins, you can use [`next-compose-plugins`](https://github.com/cyrilwanner/next-compose-plugins'). The examples in this README will follow the `next-compose-plugins` structure.
+
+```js
+// next.config.js
+const { withRasterImages, withPlayback, withSVG } = require('@moxy/next-common-plugins');
+const withPlugins = require('next-compose-plugins');
+
+module.exports = (phase, nextConfig) => {
+	withPlugins([
+		[withRasterImages()],
+		[withPlayback()],
+		[withSVG()],
+	]),
+};
+```
+
+
+## Loaders
+
+All plugins default to using [`url-loader`](https://github.com/webpack-contrib/url-loader) with the `limit` option set to `0`, which forces a fallback to [`file-loader`](https://github.com/webpack-contrib/file-loader). This means that, in practice, developers must opt in for `url-loader`'s base64 translation. Developers can choose to set a higher limit in conjunction with other rule options to accommodate the structure of their own project.
+
+With the SVG plugin, the list of loaders changes depending on whether it should produce an inline output. If *true*, the cadence of loaders, in order of execution, becomes: [`svg-css-modules-loader`](https://github.com/kevin940726/svg-css-modules-loader), [`svgo-loader`](https://github.com/rpominov/svgo-loader) and [`raw-loader`](https://github.com/webpack-contrib/raw-loader).
+
+
+## Options
+
+All plugins can be passed an options object that will spread to the `webpack` rule. With one exception, explored further below in the SVG section, you can also expect the object to spread to the `url-loader` configuration. You can refer to the [`webpack rule documentation`](https://webpack.js.org/configuration/module/#rule) and the [`url-loader documentation`](https://github.com/webpack-contrib/url-loader#readme) for more details on available options.
+
+Below you can find some common, general examples on how to use the plugins. Please refer to the section specific to each plugin further below for detailed information about each and specifics on how to use them.
+
+Excluding a directory:
+```js
+// Exclude /images/
+[withRasterImages({
+	exclude:/images\/.*$/,
```

Comment:
Can we use 4 spaces here instead of tabs just like our eslint defines? Please check all sections.

Also add a space after exclude.